api/src/main/java/net/kyori/adventure/text/ComponentCompaction.java

```diff
@@ -167,6 +167,6 @@ static Component compact(final @NotNull Component self, final @Nullable Style pa
   }
 
   private static TextComponent joinText(final TextComponent one, final TextComponent two) {
-    return Component.text(one.content() + two.content(), one.style());
+    return Component.text(one.content() + two.content(), one.style()).children(two.children());
```

Comment:
I'd prefer if this used a builder to construct the new component.

Reply:
Changed for `TextComponentImpl`